frontend/workflows/projectSelector/src/selector-reducer.tsx

```diff
@@ -27,13 +27,65 @@ const selectorReducer = (state: State, action: Action): State => {
       };
     }
     case "REMOVE_PROJECTS": {
-      // TODO: also remove any upstreams or downstreams related (only) to the project.
-      // if group == Groups.PROJECT, hide exclusive downstream upstreams
-      //
-      return {
-        ...state,
-        [action.payload.group]: _.omit(state[action.payload.group], action.payload.projects),
-      };
+      const newRemoveProjectsState = { ...state };
+
+      newRemoveProjectsState[action.payload.group] = _.omit(
+        state[action.payload.group],
+        action.payload.projects
+      );
+
+      const upstreamsToRemove = [];
+      const downstreamsToRemove = [];
+
+      // remove any upstreams or downstreams exclusive to the project from Group.UPSTREAM/Group.DOWNSTREAM
+      if (action.payload.group == Group.PROJECTS) {
+        action.payload.projects.forEach(p => {
+          _.forIn(state.upstreamToProject, (v, k) => {
+            // if an upstream is associated to the project
+            if (v[p]) {
+              // if the upstream is exclusive to project
+              if (Object.keys(v).length == 1) {
+                upstreamsToRemove.push(k);
+                // remove as it was only tied to this project and b/c both the project and upstream will be removed from the state
+                // otherwise we'll end up with stale information
+                delete state.upstreamToProject[k];
+              } else {
+                // project is not exclusive to the upstream but let's remove as it will be removed from Group.PROJECTS state
+                // otherwise we'll end up with stale information
+                delete v[p];
```

Comment:
scenario: we remove project `foo` and upstream `bar` is tied to both `foo` and `baz`. we're not going to remove the upstream `bar` but we delete {"foo": true} from "bar": {"foo": true, "baz": true} b/c we've removed `foo` from the project state. if we don't delete `foo`, then the length of the object will always be greater than 1, and if the user nexts wants to delete `baz`, the upstream will never be removed. 

```
upstreamToProject: {
  "bar": {"foo": true, "baz": true}
}
```